presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java

```diff
@@ -417,4 +418,18 @@ public void testMapToMapCast()
         assertFunction("CAST(MAP(ARRAY[1,2], ARRAY[TIMESTAMP '2016-01-02 01:02:03', TIMESTAMP '2016-02-03 03:04:05']) AS MAP(bigint, varchar))", new MapType(BIGINT, VARCHAR), ImmutableMap.of(1L, "2016-01-02 01:02:03.000", 2L, "2016-02-03 03:04:05.000"));
         assertInvalidCast("CAST(MAP(ARRAY[1, 2], ARRAY[6, 9]) AS MAP<boolean, bigint>)", "duplicate keys");
     }
+
+    @Test
+    public void testIndeterminate()
+    {
+        assertOperator(INDETERMINATE, "map(array[1,2], array[3,4])", BOOLEAN, false);
```

Comment:
@martint Do you mean we need to check whether the key is indeterminate when constructing a map? If so, I could some this constraint to map constructor.